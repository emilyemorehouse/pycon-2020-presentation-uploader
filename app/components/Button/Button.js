/* eslint-disable react/prop-types */
import React from 'react'
import { Button as GrommetButton } from 'grommet'
import styled from 'styled-components'

import { doc } from './Button.doc'
import helpers from '../../utils/helpers'

const StyledGrommetButton = styled(GrommetButton)`
  font-weight: bold;
  font-family: 'Roboto Mono';
`

/**
 *
 * Button.js
 *
 * A common button
 */
function Button({ ...rest }) {
  return <StyledGrommetButton {...rest} />
}

export default helpers.createWithDoc({
  envName: process.env.NODE_ENV,
  docFunction: doc,
  component: Button,
})
