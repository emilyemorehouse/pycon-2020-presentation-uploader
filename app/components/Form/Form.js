/* eslint-disable react/prop-types */
/**
 * Form
 */

import React from 'react'
import { Form } from 'grommet'
import { doc } from './Form.doc'
import helpers from '../../utils/helpers'

const createForm = ({
  errors,
  messages,
  onChange,
  onReset,
  onSubmit,
  value,
  children,
  style,
  ...rest
}) => (
  <Form
    errors={errors}
    messages={messages}
    onChange={onChange}
    onReset={onReset}
    onSubmit={onSubmit}
    value={value}
    style={{ maxWidth: '70vw', margin: 'auto', ...style }}
    {...rest}
  >
    {children}
  </Form>
)

export default helpers.createWithDoc({
  docFunction: doc,
  component: createForm,
})
