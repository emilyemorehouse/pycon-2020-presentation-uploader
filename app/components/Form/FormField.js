/* eslint-disable react/prop-types */
import React from 'react'
import { FormField as GrommetFormField } from 'grommet'
import styled from 'styled-components'

import { doc } from './Form.doc'
import helpers from '../../utils/helpers'

export const StyledFormField = styled(GrommetFormField)`
  label {
    margin: 15px 0 0 12px;
    font-size: 14px;
  }

  span {
    font-size: 14px;
  }
`

/**
 * FormField
 */
const FormField = props => {
  const { children, required, label, ...rest } = props
  return (
    <StyledFormField required={required} label={required ? `${label}*` : label} {...rest}>
      {children}
    </StyledFormField>
  )
}

export default helpers.createWithDoc({
  docFunction: doc,
  component: FormField,
})
