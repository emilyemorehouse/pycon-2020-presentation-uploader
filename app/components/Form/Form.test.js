import React from 'react'
import renderer from 'react-test-renderer'
import { cleanup, render, fireEvent } from '@testing-library/react'
import 'jest-styled-components'

import { Grommet } from 'grommet'
import Form from './Form'
import FormField from './FormField'
import { Button } from '../Button'

describe('Form', () => {
  afterEach(cleanup)

  test('empty', () => {
    const component = renderer.create(
      <Grommet>
        <Form />
      </Grommet>,
    )
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })

  test('with field', () => {
    const component = renderer.create(
      <Grommet>
        <Form>
          <FormField name="test" />
        </Form>
      </Grommet>,
    )
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })

  test('regexp validation', () => {
    const onSubmit = jest.fn()
    const { getByPlaceholderText, getByText, queryByText } = render(
      <Grommet>
        <Form onSubmit={onSubmit}>
          <FormField
            name="test"
            required
            validate={{ regexp: /^[a-z]/i }}
            placeholder="test input"
          />
          <Button type="submit" primary label="Submit" />
        </Form>
      </Grommet>,
    )

    fireEvent.change(getByPlaceholderText('test input'), {
      target: { value: '1' },
    })
    fireEvent.click(getByText('Submit'))
    expect(getByText('invalid')).toMatchSnapshot()

    fireEvent.change(getByPlaceholderText('test input'), {
      target: { value: 'a' },
    })
    fireEvent.click(getByText('Submit'))
    expect(queryByText('invalid')).toBeNull()
  })

  test('required validation', () => {
    const onSubmit = jest.fn()
    const { getByPlaceholderText, getByText, queryByText } = render(
      <Grommet>
        <Form onSubmit={onSubmit}>
          <FormField name="test" required placeholder="test input" />
          <Button type="submit" primary label="Submit" />
        </Form>
      </Grommet>,
    )

    fireEvent.click(getByText('Submit'))
    expect(queryByText('required')).toMatchSnapshot()
    fireEvent.change(getByPlaceholderText('test input'), {
      target: { value: '1' },
    })
    expect(queryByText('required')).toBeNull()
  })

  test('reset clears form', () => {
    const onReset = jest.fn()
    const { getByPlaceholderText, getByText, queryByText } = render(
      <Grommet>
        <Form onReset={onReset}>
          <FormField name="test" required placeholder="test input" />
          <Button type="reset" primary label="Reset" />
        </Form>
      </Grommet>,
    )
    fireEvent.change(getByPlaceholderText('test input'), {
      target: { value: 'Input has changed' },
    })
    fireEvent.click(getByText('Reset'))
    expect(queryByText('Input has changed')).toBeNull()
  })
})
