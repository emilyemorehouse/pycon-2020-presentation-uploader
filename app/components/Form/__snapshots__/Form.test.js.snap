// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Form empty 1`] = `
.c0 {
  font-size: 18px;
  line-height: 24px;
  box-sizing: border-box;
  -webkit-text-size-adjust: 100%;
  -ms-text-size-adjust: 100%;
  -moz-osx-font-smoothing: grayscale;
  -webkit-font-smoothing: antialiased;
}

<div
  className="c0"
>
  <form
    onReset={[Function]}
    onSubmit={[Function]}
    style={
      Object {
        "margin": "auto",
        "maxWidth": "70vw",
      }
    }
  />
</div>
`;

exports[`Form regexp validation 1`] = `
.c0 {
  margin-left: 12px;
  margin-right: 12px;
  margin-top: 6px;
  margin-bottom: 6px;
  font-size: 18px;
  line-height: 24px;
  color: #FF4040;
}

@media only screen and (max-width:768px) {

}

@media only screen and (max-width:768px) {

}

@media only screen and (max-width:768px) {

}

<span
  class="c0"
>
  invalid
</span>
`;

exports[`Form required validation 1`] = `
.c0 {
  margin-left: 12px;
  margin-right: 12px;
  margin-top: 6px;
  margin-bottom: 6px;
  font-size: 18px;
  line-height: 24px;
  color: #FF4040;
}

@media only screen and (max-width:768px) {

}

@media only screen and (max-width:768px) {

}

@media only screen and (max-width:768px) {

}

<span
  class="c0"
>
  required
</span>
`;

exports[`Form update 1`] = `
.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  box-sizing: border-box;
  outline: none;
  max-width: 100%;
  margin-bottom: 12px;
  min-width: 0;
  min-height: 0;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  box-sizing: border-box;
  outline: none;
  max-width: 100%;
  border-bottom: solid 1px rgba(0,0,0,0.33);
  min-width: 0;
  min-height: 0;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  box-sizing: border-box;
  outline: none;
  max-width: 100%;
  min-width: 0;
  min-height: 0;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c8 {
  display: inline-block;
  box-sizing: border-box;
  cursor: pointer;
  outline: none;
  font: inherit;
  -webkit-text-decoration: none;
  text-decoration: none;
  margin: 0;
  background: transparent;
  overflow: visible;
  text-transform: none;
  border: 2px solid #7D4CDB;
  border-radius: 18px;
  color: #444444;
  padding: 4px 22px;
  font-size: 18px;
  line-height: 24px;
  background: #7D4CDB;
  color: #f8f8f8;
  border-radius: 18px;
  -webkit-transition-property: color, background-color, border-color, box-shadow;
  transition-property: color, background-color, border-color, box-shadow;
  -webkit-transition-duration: 0.1s;
  transition-duration: 0.1s;
  -webkit-transition-timing-function: ease-in-out;
  transition-timing-function: ease-in-out;
}

.c8:hover {
  box-shadow: 0px 0px 0px 2px #7D4CDB;
}

.c3 {
  margin-left: 12px;
  margin-right: 12px;
  margin-top: 6px;
  margin-bottom: 6px;
  font-size: 18px;
  line-height: 24px;
}

.c7 {
  box-sizing: border-box;
  font-size: inherit;
  font-family: inherit;
  border: none;
  -webkit-appearance: none;
  padding: 11px;
  outline: none;
  background: transparent;
  color: inherit;
  font-weight: 600;
  margin: 0;
  border: 1px solid rgba(0,0,0,0.33);
  border-radius: 4px;
  width: 100%;
  border: none;
}

.c7::-webkit-search-decoration {
  -webkit-appearance: none;
}

.c7::-webkit-input-placeholder {
  color: #AAAAAA;
}

.c7::-moz-placeholder {
  color: #AAAAAA;
}

.c7:-ms-input-placeholder {
  color: #AAAAAA;
}

.c7::-moz-focus-inner {
  border: none;
  outline: none;
}

.c6 {
  position: relative;
  width: 100%;
}

.c0 {
  font-size: 18px;
  line-height: 24px;
  box-sizing: border-box;
  -webkit-text-size-adjust: 100%;
  -ms-text-size-adjust: 100%;
  -moz-osx-font-smoothing: grayscale;
  -webkit-font-smoothing: antialiased;
}

.c2 label {
  margin: 15px 0 0 12px;
  font-size: 14px;
}

.c2 span {
  font-size: 14px;
}

.c9 {
  font-weight: bold;
  font-family: 'Roboto Mono';
}

@media only screen and (max-width:768px) {
  .c1 {
    margin-bottom: 6px;
  }
}

@media only screen and (max-width:768px) {
  .c4 {
    border-bottom: solid 1px rgba(0,0,0,0.33);
  }
}

<div
  class="c0"
>
  <form
    style="max-width: 70vw; margin: auto;"
  >
    <div
      class="c1 c2"
    >
      <label
        class="c3"
      >
        undefined*
      </label>
      <div
        class="c4"
      >
        <div
          class="c5"
        >
          <div
            class="c6"
          >
            <input
              autocomplete="off"
              class="c7"
              name="test"
              placeholder="test input"
              value=""
            />
          </div>
        </div>
      </div>
    </div>
    <div
      class="c1 c2"
    >
      <div
        class="c4"
      >
        <div
          class="c5"
        >
          <div
            class="c6"
          >
            <input
              autocomplete="off"
              class="c7"
              name="test2"
              placeholder="test-2 input"
              value=""
            />
          </div>
        </div>
      </div>
    </div>
    <button
      class="c8 c9"
      type="submit"
    >
      Submit
    </button>
  </form>
</div>
`;

exports[`Form with field 1`] = `
.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  box-sizing: border-box;
  outline: none;
  max-width: 100%;
  margin-bottom: 12px;
  min-width: 0;
  min-height: 0;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  box-sizing: border-box;
  outline: none;
  max-width: 100%;
  border-bottom: solid 1px rgba(0,0,0,0.33);
  min-width: 0;
  min-height: 0;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  box-sizing: border-box;
  outline: none;
  max-width: 100%;
  min-width: 0;
  min-height: 0;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c6 {
  box-sizing: border-box;
  font-size: inherit;
  font-family: inherit;
  border: none;
  -webkit-appearance: none;
  padding: 11px;
  outline: none;
  background: transparent;
  color: inherit;
  font-weight: 600;
  margin: 0;
  border: 1px solid rgba(0,0,0,0.33);
  border-radius: 4px;
  width: 100%;
  border: none;
}

.c6::-webkit-search-decoration {
  -webkit-appearance: none;
}

.c6::-webkit-input-placeholder {
  color: #AAAAAA;
}

.c6::-moz-placeholder {
  color: #AAAAAA;
}

.c6:-ms-input-placeholder {
  color: #AAAAAA;
}

.c6::-moz-focus-inner {
  border: none;
  outline: none;
}

.c5 {
  position: relative;
  width: 100%;
}

.c0 {
  font-size: 18px;
  line-height: 24px;
  box-sizing: border-box;
  -webkit-text-size-adjust: 100%;
  -ms-text-size-adjust: 100%;
  -moz-osx-font-smoothing: grayscale;
  -webkit-font-smoothing: antialiased;
}

.c2 label {
  margin: 15px 0 0 12px;
  font-size: 14px;
}

.c2 span {
  font-size: 14px;
}

@media only screen and (max-width:768px) {
  .c1 {
    margin-bottom: 6px;
  }
}

@media only screen and (max-width:768px) {
  .c3 {
    border-bottom: solid 1px rgba(0,0,0,0.33);
  }
}

<div
  className="c0"
>
  <form
    onReset={[Function]}
    onSubmit={[Function]}
    style={
      Object {
        "margin": "auto",
        "maxWidth": "70vw",
      }
    }
  >
    <div
      className="c1 c2"
      onBlur={[Function]}
      onFocus={[Function]}
    >
      <div
        className="c3"
      >
        <div
          className="c4"
        >
          <div
            className="c5"
          >
            <input
              autoComplete="off"
              className="c6"
              name="test"
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              value=""
            />
          </div>
        </div>
      </div>
    </div>
  </form>
</div>
`;
